src/compiler/validation.jl

```diff
@@ -489,7 +493,11 @@ function check_ir!(@nospecialize(job::CompilerJob), errors::Vector{IRError}, imp
                     newf, _ = get_function!(mod, fname, FT)
                 else
                     found = nothing
-                    for lbb in blocks(initfn), linst in collect(instructions(lbb))
+                    for lbb in blocks(initfn)
+                    liter = LLVM.API.LLVMGetFirstInstruction(lbb)
+                    while liter != C_NULL
```

Comment:
Why not simply `for linst in instructions(lbb)`?

Reply:
hm will look at, elsewhere it is required because we may delete the present instruction (so this is akin to the pre increment iterator in llvm)